FreeAPSWatch WatchKit Extension/Views/NutrientsView.swift

```diff
@@ -0,0 +1,192 @@
+        let label: String
+        let labelAlignment: Alignment
+        let max: Int
+        let accentColor: Color
+
+        var fontForSizeCategory: Font {
+            switch sizeCategory {
+            case .extraSmall,
+                 .small:
+                return .system(size: 12)
+            case .medium:
+                return .system(size: 14)
+            case .large:
+                return .system(size: 16)
+            default:
+                return .system(size: 20)
+            }
+        }
+
+        var fontForSizeCategoryForHundreds: Font {
+            switch sizeCategory {
+            case .extraSmall:
+                return .system(size: 8)
+            case .small:
+                return .system(size: 9)
+            case .large,
+                 .medium:
+                return .system(size: 14)
+            default:
+                return .system(size: 16)
+            }
+        }
+
+        var body: some View {
+            GeometryReader { geometry in
+                Picker(selection: $selection) {
+                    ForEach(0 ... (max / step), id: \.self) {
+                        let value = $0 * step
+                        Text("\(value) g")
+                            .tag(value)
+                            .font(value < 99 ? fontForSizeCategory : fontForSizeCategoryForHundreds)
+                            .foregroundColor(accentColor)
+                    }
+                } label: {
+                    var labelFrameTranslationX: CGFloat {
+                        let offset = geometry.size.width + Constants.pickerHSpacing
+                        if labelAlignment == .leading { return offset }
+                        if labelAlignment == .trailing { return -offset }
+                        return 0
+                    }
+                    Text(label)
+                        .foregroundColor(.black)
+                        .padding(.vertical, 2)
+                        .padding(.horizontal, 4)
+                        .background(
+                            RoundedRectangle(cornerRadius: 6)
+                                .fill(accentColor)
+                        )
+                        .textCase(.uppercase)
+                        .frame(width: labelFrameWidth, alignment: labelAlignment)
+                        .transformEffect(.init(translationX: labelFrameTranslationX, y: 0))
+                }
+                .pickerStyle(.wheel)
+                .frame(maxWidth: geometry.size.width)
+                .onTapGesture(count: 2, perform: handlePickerDoubleTapGesture)
+            }
+        }
+
+        private func handlePickerDoubleTapGesture() {
+            step = step == Constants.defaultPickerStep ? 1 : Constants.defaultPickerStep
+            if step == Constants.defaultPickerStep {
+                selection = Int(Double(selection / Constants.defaultPickerStep).rounded(.down)) * Constants
+                    .defaultPickerStep
+            }
+        }
```

Comment:
Double tap changes Picker step from 5 to 1 and vice versa. When the `step` is 5 we round the Picker value to the lower closest one divisible by 5.
An example: double tapping on a Picker when its value is `77` will round it to `75`